src/main/java/kitchenpos/menu/domain/Menu.java

```diff
@@ -0,0 +1,88 @@
+package kitchenpos.menu.domain;
+
+import kitchenpos.common.domain.Price;
+import kitchenpos.menugroup.domain.MenuGroup;
+
+import javax.persistence.Column;
+import javax.persistence.Embedded;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.ManyToOne;
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Objects;
+
+@Entity
+public class Menu {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Column(nullable = false)
+    private String name;
+
+    @Embedded
+    @Column(nullable = false)
+    private Price price;
+
+    @ManyToOne
+    private MenuGroup menuGroup;
+
+    @Embedded
+    private MenuProducts menuProducts;
+
+    protected Menu() {
+    }
+
+    public Menu(String name, BigDecimal price, MenuGroup menuGroup, MenuProducts menuProducts) {
+        validatePrice(new Price(price), menuProducts);
+
+        this.name = name;
+        this.price = new Price(price);
+        this.menuGroup = menuGroup;
+        this.menuProducts = menuProducts;
+        menuProducts.changeMenu(this);
+    }
+
+    public Long getId() {
+        return id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public Price getPrice() {
+        return price;
+    }
+
+    public MenuGroup getMenuGroup() {
+        return menuGroup;
+    }
+
+    public List<MenuProduct> getMenuProducts() {
+        return menuProducts.getMenuProducts();
+    }
+
+    private void validatePrice(Price price, MenuProducts menuProducts) {
+        if (!price.isEqual(menuProducts.calculateSum())) {
```

Comment:
```suggestion
        if (!price.isEqual(menuProducts.totalPrice())) {
```
와 같은 함수명이 더 어울릴거 같은데, 어떻게 생각하시나요?

Reply:
totalPrice()가 좀 더 직관적인 이름인 것 같습니다! 👍
작명이 제일 어려운 것 같아요 😭